test/specs/commonTests.js

```diff
@@ -396,6 +402,72 @@ export const implementsAlignedProp = (Component, requiredProps = {}) => {
   })
 }
 
+export const implementsIconProp = (Component, requiredProps = {}) => {
+  const iconClass = faker.hacker.noun()
+  const assertValid = (wrapper) => {
+    wrapper.should.have.className('icon')
+    wrapper.should.have.descendants('Icon')
+    wrapper.find('Icon')
+      .should.have.className(iconClass)
+  }
+
+  describe('icon (common)', () => {
+    _noDefaultClassNameFromProp(Component, 'icon')
+
+    it('has no i when not defined', () => {
+      shallow(<Component />)
+        .should.not.have.descendants('i')
+    })
+
+    it('adds a i as first child', () => {
+      shallow(<Component icon={iconClass} />)
+        .childAt(0)
+        .should.match('i')
+    })
+
+    it('accepts an Icon instance', () => {
+      const icon = <Icon className={iconClass} />
+      assertValid(shallow(<Component icon={icon} />))
+    })
+
+    it('accepts an icon className string', () => {
+      assertValid(shallow(<Component icon={iconClass} />))
+    })
+  })
+}
```

Comment:
👍 thanks!  some of these will need to be relaxed to pass on other components.  we'll leave them for now and address that later.